### PR TITLE
Dog owner profile

### PR DESCRIPTION
**Pull request by nicknytko**
profile screen for dog owners & dog, also very placeholder dog icon on the map to get to the profile screen.

screen shot 2018-11-23 at 12 24 54 pm

screen shot 2018-11-23 at 12 25 05 pm

screen shot 2018-11-23 at 12 25 09 pm

### CODE REVIEW

File: app/src/main/java/cs465/illinois/edu/dogonthequad/DogOwnerProfileActivity.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import android.os.Bundle;
+import android.widget.ExpandableListView;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class DogOwnerProfileActivity extends Activity {
+
+    ExpandableListAdapter listAdapter;
```
Comment (divy4) [inline app/src/main/java/cs465/illinois/edu/dogonthequad/DogOwnerProfileActivity.java]: You probably want to specify these as `private`.

File: app/src/main/java/cs465/illinois/edu/dogonthequad/DogOwnerProfileActivity.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        // preparing list data
+        prepareListData();
+
+        listAdapter = new ExpandableListAdapter(this, listDataHeader, listDataChild);
+
+        // setting list adapter
+        expListView.setAdapter(listAdapter);
+    }
+
+    private void prepareListData() {
```
Comment (divy4) [inline app/src/main/java/cs465/illinois/edu/dogonthequad/DogOwnerProfileActivity.java]: Nice job of extracting data collection from `onCreate`.

File: app/src/main/java/cs465/illinois/edu/dogonthequad/ExpandableListAdapter.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+import android.content.Context;
+import android.graphics.Typeface;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseExpandableListAdapter;
+import android.widget.TextView;
+
+public class ExpandableListAdapter extends BaseExpandableListAdapter {
```
Comment (divy4) [inline app/src/main/java/cs465/illinois/edu/dogonthequad/ExpandableListAdapter.java]: What is the purpose of this class? (This isn't a criticism, by the way. I'm just not familiar with coding expandable lists.)

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -24,5 +24,10 @@
     <string name="placeholder_num">69</string>
     <string name="return_to_edit">Confirm Change</string>
     <string name="social_level">Social Level:</string>
+    <string name="placeholder_profile_name">Donnie Doggo</string>
```
Comment (divy4) [inline app/src/main/res/values/strings.xml]: Please sort these values (so we don't create duplicates in other PR's).

File: app/src/main/java/cs465/illinois/edu/dogonthequad/ExpandableListAdapter.java (hunk 3 above)
Comment (nicknytko) [reply app/src/main/java/cs465/illinois/edu/dogonthequad/ExpandableListAdapter.java]: The general standard in android for lists (expandable lists, recycler views, etc.) is to have an abstract base class and the app developer implements an adapter for it, which can handle things like populating list entries with data, touch events, etc.

Comment (divy4) [reply app/src/main/java/cs465/illinois/edu/dogonthequad/ExpandableListAdapter.java]: Thanks!

**Pull request merged**